### PR TITLE
Upgrade the version of ecto

### PR DESCRIPTION
**Pull request by tpitale**
* Update the version of ecto and fix the dependency issues with Bcrypt
* Fixing tests

Completes the work started by @elpic in PR #86. That PR appears to be abandoned since March. I asked about it in May. It is now into September.

### CODE REVIEW

**Comment by tpitale** [comment]
Removed elixir 1.5 from travis config, added 1.8, 1.9.

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,16 @@
 dist: trusty
 sudo: false
 language: elixir
-elixir: '1.7.3'
-otp_release: '21.1.1'
+elixir: '1.9.1'
+otp_release: '22.0.7'
 matrix:
   include:
+    - elixir: '1.8.1'
```
Comment (mustafaturan) [inline .travis.yml]: Can you re-order alpha-numerically the list? And include multiple OTP versions when possible?

Comment (tpitale) [reply .travis.yml]: Do you have a list of OTP version you want to use? I couldn't find a reliable list of Elixir version + OTP version.

Comment (tpitale) [reply .travis.yml]: I used this https://github.com/elixir-lang/elixir/blob/master/lib/elixir/pages/Compatibility%20and%20Deprecations.md#compatibility-between-elixir-and-erlangotp

Let me know @mustafaturan if you'd like me to use a different range.

Comment (mustafaturan) [reply .travis.yml]: Lgtm

**Pull request merged**